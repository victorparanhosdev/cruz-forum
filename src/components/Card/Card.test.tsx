import { render, screen } from '@testing-library/react'
import { Card } from './'

describe('<Card/>', () => {
  const mockCardId = '123'

  it('deverá renderizar o card e receber o ID do Card', () => {
    render(<Card data-testid="custom-element" cardId={mockCardId} />)
    const element = screen.getByTestId('custom-element')
    const link = screen.getByRole('link')

    expect(link).toHaveAttribute('href', `/topicos/123`)
    expect(element).toBeInTheDocument()
  })

  it('receberá uma nova className', () => {
    render(
      <Card
        className="bg-black"
        data-testid="custom-element"
        cardId={mockCardId}
      />,
    )
    const element = screen.getByTestId('custom-element')

    expect(element).toHaveClass('bg-black')
  })

  it('deve corresponder ao snapshot', () => {
    const { asFragment } = render(<Card cardId={mockCardId} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
