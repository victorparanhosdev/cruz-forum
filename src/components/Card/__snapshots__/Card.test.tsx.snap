// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> deve corresponder ao snapshot 1`] = `
<DocumentFragment>
  <a
    href="/topicos/123"
  >
    <div
      class="flex cursor-pointer gap-2.5 rounded-lg border border-white px-3 py-2 hover:border-green-200 hover:bg-hover-btn-gray"
    >
      <img
        alt="Imagem do perfil"
        class="size-8 rounded-full object-cover"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fgithub.com%2Fvictorparanhosdev.png&w=64&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Fvictorparanhosdev.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Fvictorparanhosdev.png&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <div>
        <p
          class="text-sm font-medium"
        >
          Título do Tópico
        </p>
        <span
          class="flex items-center gap-1 text-[10px] text-gray-400"
        >
          <svg
            fill="currentColor"
            height="16"
            viewBox="0 0 256 256"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M232,56A104.11,104.11,0,0,1,128,160H51.31l34.35,34.34a8,8,0,0,1-11.32,11.32l-48-48a8,8,0,0,1,0-11.32l48-48a8,8,0,0,1,11.32,11.32L51.31,144H128a88.1,88.1,0,0,0,88-88,8,8,0,0,1,16,0Z"
            />
          </svg>
           comentado há 6 minutos
        </span>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
